YoCode/Program.cs

```diff
@@ -2,14 +2,23 @@
 using System.Linq;
 using System.IO;
 using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
 
 namespace YoCode
 {
     public static class Program
     {
+        public static IConfiguration Configuration;
+
+        static string CMDToolsPath;
 
         static void Main(string[] args)
         {
+            //This needs to be implemented in order for DuplicationCheck to work
+            var builder = new ConfigurationBuilder().SetBasePath(Path.GetFullPath(@"..\..\..\Properties")).AddJsonFile("launchsettings.json");
```

Comment:
The `@"..\..\..\Properties"` part makes me a feel a bit 😢 . It will be okay during development but won't work when we ship the tool outside the team. Don't do anything now though, if this works for the team, just leave it. I'll have a think about a better solution.

Reply:
Yeah, that's where the .json file was stored though, couldn't think of a better solution at a time

Reply:
I've been trying things out and I have 2 suggestions.
1. We should call the file `appsettings.json` instead of `launchsettings.json`. I think `launchsettings.json` has a meaning in .Net Core and so it expects certain things to be in there, like the name of the command to execute, which don't apply to our project (we're not using ASP.Net). If we call it `appsettings.json` we can call it whatever we like and just have the things we want, no extra options.
2. Put it in the root of the project, not in properties, and then get the build to copy it to the right place. That means we can avoid the `@"..\..\..\Properties"` thing. Like this:
![appsettings_copy_on_build](https://user-images.githubusercontent.com/1519910/42762828-3ddff2b4-8909-11e8-8e9d-8dea65e36fbc.gif)

My `appsettings.json` just has this in it, nothing else:
```
{
  "duplicationCheckSetup": {
    "CMDtoolsDir": "C:\\tools\\ReSharperCLT"
  }
}
```
It seems a bit cleaner to me.